src/python/foglamp/data_purge/purge.py

```diff
@@ -0,0 +1,332 @@
+                                                          default=_READING_TABLE.name, primary_key=True),
+                                        sqlalchemy.Column('start_time', sqlalchemy.VARCHAR(255),
+                                                          default=sqlalchemy.func.current_timestamp),
+                                        sqlalchemy.Column('end_time', sqlalchemy.VARCHAR(255),
+                                                          default=sqlalchemy.func.current_timestamp),
+                                        sqlalchemy.Column('total_rows_removed', sqlalchemy.INTEGER, default=0),
+                                        sqlalchemy.Column('total_rows_remaining', sqlalchemy.INTEGER, default=0),
+                                        sqlalchemy.Column('total_unsent_rows', sqlalchemy.INTEGER, default=0),
+                                        sqlalchemy.Column('total_failed_to_remove', sqlalchemy.INTEGER, default=0))
+
+"""Methods that support the purge process.  
+"""
+
+
+def convert_timestamp(set_time: str) -> datetime.timedelta:
+    """Convert "age" in config file to timedelta. If only an integer is specified,  then 
+        the code assumes that it is already in minutes (ie age:1 means wait 1 minute) 
+    Args:
+        set_time (str): Newest amount of  time back to delete
+    Returns:
+        converted set_time to datetime.timedelta value
+    """
+    if type(set_time) is int or set_time.isdigit():
+        return datetime.timedelta(minutes=int(set_time))
+    time_dict = {}
+    tmp = 0
+
+    for value in set_time.split(" "):
+        if value.isdigit() is True:
+            tmp = int(value)
+        else:
+            time_dict[value] = tmp
+
+    time_in_sec = datetime.timedelta(seconds=0)
+    time_in_min = datetime.timedelta(minutes=0)
+    time_in_hr = datetime.timedelta(hours=0)
+    time_in_day = datetime.timedelta(days=0)
+
+    for key in time_dict.keys():
+        if 'sec' in key:
+            time_in_sec = datetime.timedelta(seconds=time_dict[key])
+        elif 'min' in key:
+            time_in_min = datetime.timedelta(minutes=time_dict[key])
+        elif ('hr' in key) or ('hour' in key):
+            time_in_hr = datetime.timedelta(hours=time_dict[key])
+        elif ('day' in key) or ('dy' in key):
+            time_in_day = datetime.timedelta(days=time_dict[key])
+    return time_in_sec+time_in_min+time_in_hr+time_in_day
+
+
+def execute_command_with_return_value(stmt: str) -> dict:
+    """Imitate connection to postgres that returns result.    
+    Args:
+        stmt (str): generated SQL query   
+    Returns:
+        Returns result set 
+    """
+    query_result = _CONN.execute(stmt)
+    return query_result.fetchall()
+
+
+def execute_command_without_return_value(stmt: str) -> None:
+    """Imitate connection to Postgres and a query that doesn't generate results
+    Args:
+        stmt (str): DELETE stmt 
+    """
+    _CONN.execute(stmt)
+    _CONN.execute("commit")
```

Comment:
By default in SQLAlchamy all data changing operations are autocommit http://docs.sqlalchemy.org/en/latest/core/connections.html#understanding-autocommit